src/Illuminate/Database/Query/Builder.php

```diff
@@ -748,6 +748,8 @@ public function whereIn($column, $values, $boolean = 'and', $not = false)
         // execute those Closures, then we can re-construct the entire sub-selects.
         if ($values instanceof Closure) {
             return $this->whereInSub($column, $values, $boolean, $not);
+        } elseif ($values instanceof Collection) {
+            return $this->whereIn($column, $values->toArray(), $boolean, $not);
```

Comment:
My thought was that if something happens before the condition, the method needs to be re-called. I should probably change this for now though.